File: frontend/src/api/aiInterpretation.ts
import { fetchWithAuth } from "./fetchWithAuth";

export type InterpretSignResponse = {
  sign_id: number;
  markdown: string;
  note: string;
};

export async function interpretSign(signId: number): Promise<InterpretSignResponse> {
  const response = await fetchWithAuth("/ai/interpret-sign", {
    method: "POST",
    headers: {
      "Content-Type": "application/json"
    },
    body: JSON.stringify({ sign_id: signId })
  });

  if (!response.ok) {
    const data = await response.json().catch(() => null);
    throw new Error(data?.detail || "AI 解签失败，请稍后再试");
  }

  return response.json();
}

File: backend/app/schemas/ai_interpretation.py
from pydantic import BaseModel, Field


class InterpretSignRequest(BaseModel):
    sign_id: int = Field(gt=0)


class InterpretSignResponse(BaseModel):
    sign_id: int
    markdown: str
    note: str = "以下内容为传统文化角度的解读与启发，不代表现实预测。"
